### PR TITLE
feat : add login, logout, signup functionality

### PR DESCRIPTION
**Pull request by Jay-Gadhiya**
Test link: https://silk-shopy-ecommerce-4d3l5l5l8-jay-gadhiya.vercel.app/

Add functionality: Signup, Login, Logout, guest login credential
folder changes: Authentication, Navbar, Contexts.
context : authentication-context.jsx
reducer : auth-reducer

### CODE REVIEW

File: silkshopy-ecommerce/src/pages/Authentication/Login.jsx [hunk 1]
```diff
@@ -1,22 +1,68 @@
...
+
+    // User input values click handler
+    const addUserEmail = (e) => {
+        setUserData((pre) => ({ ...pre, email : e.target.value }))
+    }
+
+    const addUserPassword = (e) => {
+        setUserData((pre) => ({ ...pre, password : e.target.value }))
+    }
+
```
Comment (Kevin-Solomon) [inline silkshopy-ecommerce/src/pages/Authentication/Login.jsx]: You can use one function to update both ```email``` and ```password```.
Use ```[e.target.name] : e.target.value```

**Review by Kevin-Solomon — APPROVED** [review]

File: silkshopy-ecommerce/src/pages/Authentication/Login.jsx (hunk 1 above)
Comment (Jay-Gadhiya) [reply silkshopy-ecommerce/src/pages/Authentication/Login.jsx]: yes, i will do that

File: silkshopy-ecommerce/src/backend/db/users.js [hunk 2]
```diff
@@ -13,7 +13,7 @@ export const users = [
     firstName: "Adarsh",
     lastName: "Balika",
     email: "adarshbalika@gmail.com",
-    password: bcyrpt.hashSync("adarshBalika123", 5),
+    password: "adarshBalika123",
```
Comment (AtulPant2704) [inline silkshopy-ecommerce/src/backend/db/users.js:16]: Don't remove encrypted password, you should have your passwords encypted only.

**Review by AtulPant2704 — APPROVED** [review]

File: silkshopy-ecommerce/src/backend/db/users.js (hunk 2 above)
Comment (Jay-Gadhiya) [reply silkshopy-ecommerce/src/backend/db/users.js:16]: yes , i will add that back

**Pull request merged**